x-pack/plugins/search_playground/server/lib/conversational_chain.ts

```diff
@@ -236,6 +236,10 @@ class ConversationalChainFn {
                   type: 'prompt_token_count',
                   count: getTokenEstimateFromMessages(msg),
                 });
+                data.appendMessageAnnotation({
+                  type: 'search_query',
+                  question,
```

Comment:
This needs a tweak - the question const here is always the last message submission which isn't the question used for the search. You want to modify the question standalone chain to capture the transformed question that was returned by the LLM. The runnable sequence runs each function in succession. We want to insert this runnable after the LLM model has been called.

For example:

```
standaloneQuestionChain = RunnableSequence.from([
        {
          context: (input) => input.chat_history,
          question: (input) => input.question,
        },
        questionRewritePromptTemplate,
        this.options.model,
        (inputs) => {
          data.appendMessageAnnotation({
            type: 'search_query',
            question: inputs.question,
          });
          return inputs;
        },
        new StringOutputParser(),
      ])
```

Reply:
you should get something similar and not be exactly the same response as the last message

![image](https://github.com/user-attachments/assets/14b963f0-5318-4fbb-94d1-137aebb8e660)